src/status_im/ui/screens/profile/contact/views.cljs

```diff
@@ -55,22 +65,35 @@
                  :selectable          true}
      value]]])
 
-(defn profile-info-contact-code-item [public-key]
-  [profile-info-item
-   {:label               (i18n/label :t/contact-code)
-    :accessibility-label :profile-public-key
-    :value               public-key}])
+(defn render-detail [{:keys [name public-key address] :as detail}]
+  [list-item/list-item
+   {:type     :small
+    :title    name
+    :subtitle (utils/get-shortened-address public-key)
+    :image    [chat-icon/contact-icon-contacts-tab detail]
+    :accessories [[vector-icons/icon :main-icons/share styles/contact-profile-detail-share-icon-color]]}])
 
-(defn profile-info [{:keys [public-key]}]
+(defn profile-details-list-view [details]
+  [list/flat-list {:data                    details
+                   :default-separator?      true
+                   :key-fn                  :public-key
+                   :render-fn               render-detail}])
+
+(defn- contact->details
+  "prepare a sequence of datails from a given contact"
+  [contact]
```

Comment:
could you elaborate?

Reply:
@flexsurfer the function should "prepare" a list of details (accounts\wallet) from the current contact. Since i don't know how will actually wallet\accounts be stored in db for a given contact, I prepared a function for the purpose.
Currently it just create an array with current account.

Reply:
contact will always be a map, wallet and accounts will be part of it. I'd just remove this fn a use contact directly

Reply:
@yenda I missed this comment, later this afternoon i'll remove it.